File: rb/features.rb
# NdbcBuoyData SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module NdbcBuoyDataFeatures
  def self.make_feature(name)
    case name
    when "base"
      NdbcBuoyDataBaseFeature.new
    when "test"
      NdbcBuoyDataTestFeature.new
    else
      NdbcBuoyDataBaseFeature.new
    end
  end
end

File: py/ndbcbuoydata_sdk/features.py
# NdbcBuoyData SDK feature factory

from ndbcbuoydata_sdk.feature.base_feature import NdbcBuoyDataBaseFeature
from ndbcbuoydata_sdk.feature.test_feature import NdbcBuoyDataTestFeature


def _make_feature(name):
    features = {
        "base": lambda: NdbcBuoyDataBaseFeature(),
        "test": lambda: NdbcBuoyDataTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
